npm/private/npm_translate_lock.bzl

```diff
@@ -132,35 +133,87 @@ def _gather_values_from_matching_names(keyed_lists, *names):
                 result.append(v)
     return result
 
-def _get_npm_auth(rctx):
-    _NPM_TOKEN_KEY = "//registry.npmjs.org/:_authtoken"
-    token = None
+def get_npm_auth(npmrc, npmrc_path, environ):
+    """Parses npm tokens, registries and scopes from `.npmrc`.
+
+    - creates a token by registry dict: {registry: token}
+    - creates a registry by scope dict: {scope: registry}
+
+    For example:
+        Given the following `.npmrc`:
+
+        ```
+        @myorg:registry=https://somewhere-else.com/myorg
+        @another:registry=https://somewhere-else.com/another
+        ; would apply only to @myorg
+        //somewhere-else.com/myorg/:_authToken=MYTOKEN1
+        ; would apply only to @another
+        //somewhere-else.com/another/:_authToken=MYTOKEN2
+        ```
+
+        `get_npm_auth(rctx)` creates the following dict:
+
+        ```starlark
+        tokens = {
+            "somewhere-else.com/myorg": "MYTOKEN1",
+            "somewhere-else.com/another": "MYTOKEN2",
+        }
+        registries = {
+                "@myorg": "somewhere-else.com/myorg",
+                "@another": "somewhere-else.com/another",
+        }
+        auth = (tokens, registries)
+        ```
 
-    # Read token from npmrc label
-    if rctx.attr.npmrc:
-        npmrc_path = rctx.path(rctx.attr.npmrc)
-        npmrc = parse_ini(rctx.read(npmrc_path))
-        if _NPM_TOKEN_KEY in npmrc:
-            # Parse environment variable from config
-            token = npmrc[_NPM_TOKEN_KEY]
+    Args:
+        npmrc: The `.npmrc` file.
+        npmrc_path: The file path to `.npmrc`.
+        environ: A map of environment variables with their values.
+
+    Returns:
+        A list with a tokens dict and a registries dict.
```

Comment:
nit: A tuple with a tokens dict and a registries dict

Reply:
Same as above. On it ⛵